qemu/tests/qemu_guest_agent.py

```diff
@@ -3744,6 +3745,111 @@ def __init__(self, test, params, env):
         self.qemu_ga_pkg = params.get("qemu_ga_pkg", "")
         self.gagent_src_type = params.get("gagent_src_type", "url")
 
+    @error_context.context_aware
+    def _check_serial_driver(self, test, params, env):
+        '''
+        Check whether serial device is installed or not, if not install it.
+        Test steps:
+        1) Check whether serial driver is running.
+        2) Install serial driver.
+        3) Reboot vm
+        4) Verify serial has been installed.
+
+        :param test: QEMU test object
+        :param params: Dictionary with the test parameters
+        :param env: Dictionary with test environment
+        '''
+
+        def _chk_cert(session, cert_path):
+            chk_cmd = "certutil -verify %s"
+            chk_cmd %= cert_path
+            # it may take a while to verify cert file so lets wait a little longer
+            out = session.cmd(chk_cmd, timeout=360)
+            if re.search("Expired certificate", out, re.I):
+                logging.warning("Certificate '%s' is expired!", cert_path)
+            if re.search("Incomplete certificate chain", out, re.I):
+                logging.warning("Incomplete certificate chain! Details:\n%s", out)
+
+        def _add_cert(session, cert_path, store):
+            add_cmd = "certutil -addstore -f %s %s"
+            add_cmd %= (store, cert_path)
+            session.cmd(add_cmd, timeout=120)
+
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+
+        driver_name = params["driver_name_serial"]
+        device_hwid = params["device_hwid_serial"]
+        media_type = params["virtio_win_media_type"]
+
+        error_context.context("Check whether serial drive is running.",
+                              logging.info)
+        if params["gagent_serial_type"] == "virtio":
+            check_serial_driver = session.cmd_output(params["cmd_check_serial"]
+                                                     ).strip()
```

Comment:
And here:
```suggestion
            check_serial = params.get("cmd_check_serial")
            if check_serial:
              check_serial_result = session.cmd_output(check_serial).strip()
```
However, feel free to keep it if you would like to.